### ISSUE TITLE
Starts doesn't behave as expected

### ISSUE THREAD

**Issue by buffpojken** [rossta/montrose#42]
When using hash-notation to create a recurrence, starts doesn't seem to behave as expected, see attached example. 

```
schedule = Montrose.r(every: :week, on: "tuesday", at: "5:00")

old_dates = schedule.take(5)

new_schedule = schedule.merge(starts: "2016-06-23")

new_dates = new_schedule.take(5)

puts old_dates == new_dates

third_schedule = Montrose.r(every: :week, on: "tuesday", at: "5:00", starts: "2016-06-23")

third_dates = third_schedule.take(5)

puts third_dates == old_dates
```

Shouldn't both the #merge and the new schedule with starts return dates after 2016-06-23?

**Comment by rossta** [comment]
Yes, you're right, I agree this is a bug. Currently, [`:at` overrides the `:starts`](https://github.com/rossta/montrose/blob/f94937406c5673799e9fac65219d9f9bd76ea85e/lib/montrose/options.rb#L197) option so trying to set `:starts` to something else isn't working as expected. The `:at` option does imply a `:starts` time when none is given but it should also take an existing `:starts` option into account.

**Comment by buffpojken** [comment]
Hmm, yes - I see what you mean. 

I'll see if I have time to PR that tonight or tomorrow.

**Comment by buffpojken** [comment]
What I think should be a more obvious behaviour is the following:

`:at` should be used to set hour/minute in terms of when within a given day an event occurs but
`:starts` should be used to set a starting date. 

The combination of date + time within the scope of `:starts` doesn't really make sense since the relevancy of the time-part is highly contextual on the other set of rules within a given recurrence context, and no expressiveness is lost by separating those into two separate actions, i.e. `{:starts => "2016-12-31", :at => "5 am"}` is equivalent with `{:at => "2016-12-21 5 am"}`. 

Basically, what I'm proposing is that `:at` should ignore (and not set) any date parts while `:starts` should ignore (and not set) any time parts provided. 

Thoughts?

**Comment by buffpojken** [comment]
Hmm, follow-up thought: 

Given the behaviour described above - `:starts` as a term should maybe be renamed to `:from` or similar?

**Comment by rossta** [comment]
Thanks for your ideas, I'd have to give this some thought. Treating `:starts` as only the date parts could be plausible but I want to consider how this affects other use cases. This all touches on a semantics issue I noted in #6; `:starts` is playing dual roles as either the actual starting point of the recurrence and, in other cases, just a lower bound, like a `:from` or `:after` option would suggest.

**Comment by rossta** [comment]
@buffpojken I opened #43 as a way to reconcile that `:starts` acts like a date or a datetime depending on the context. Now when using `:at`, the recurrence will start at the first `:at` time of day on the date represented in the `:starts` timestamp. Let me know what you think, otherwise, I'll merge in the next day or so.

**Comment by buffpojken** [comment]
I think it looks splendid :) 

I'll test it out in our project once it's merged but it sure looks fine to me!
